buildtest/builders/spack.py

```diff
@@ -241,6 +241,10 @@ def _spack_environment(self, spack_env):
             spack_env_create_line = " ".join(cmd)
             lines += [spack_env_create_line]
 
+            # deactivate environment ('spack env deactivate')
+            if spack_env.get("deactivate"):
+                cmd = ["spack env deactivate"]
```

Comment:
@Mariamajib just change this variable `cmd` to `lines += ['spack env deactivate']` because thats the variable that contains list of commands. After this if condition the value is no longer retained since i realize we use `cmd` variable in the later sections